etc/sds_operations_gluster-3.8.3.yaml

```diff
@@ -1,160 +1,543 @@
 ---
-sds_version: gluster-3.8.3
+tendrl_schema_version: 0.1
+
 valid_objects:
+- tendrl_context
 - volume
 - brick
 - peer
-- snapshot
 
 object_details:
+  tendrl_context:
+    enabled: True
+    value: /clusters/$cluster.id/tendrl_context
+    attrs:
+      sds_type:
+        type: String
+        help: "gluster-3.8.3"
   volume:
+    enabled: True
+    value: "/clusters/$cluster.id/volumes/$volume.id"
     attrs:
       volname:
         type: String
+        help: "Name of gluster volume"
       stripe_count:
         type: Integer
+        help: "Stripe count of volume"
       replica_count:
         type: Integer
+        help: "Replica count of volume"
       disperse_count:
         type: Integer
+        help: "Disperse count of volume"
       redundancy_count:
         type: Integer
+        help: "Redundancy count of volume"
       transport:
         type: String
+        help: "Transport type for volume"
       brickdetails:
-        type: list[brick]
+        type: brick[]
+        help: "List of bricks for volume"
       force:
         type: Boolean
-      directory:
-        type: String
-      bitrot:
-        type: Boolean
-      scrub-frequency:
-        type: Integer
-      scrub-throttle:
-        type: Integer
+        help: "If force execute the action"
 
     atoms:
-        info:
-          type: "get"
         create:
-          run: "gluster_bridge.atoms.volume.create"
+          enabled: True
+          uuid: "242f6190-9b37-11e6-950d-a24fc0d9649c"
+          flows: ["CreateVolume"]
```

Comment:
Do you to have flows like CreateDistributeVolume, CreateStripedVolume, CreateStripedReplicateVolume etc as separate flows?
Based on non-mandatory fields possible for gluster volumes this permutation is going to be lot big in no I feel. Dont you feel we should some mechanism to mention optional arguments and only if they are passed we should invoke flow with different arguments?

@brainfunked @nthomas-redhat @nnDarshan @anmolbabu thoughts?
I strongly feel we should have some mechanism to mention a optional list of arguments as well for flows and this is the right juncture when we are defining the structure of yaml.